fluent_pose_synthesis/data/load_data.py

```diff
@@ -0,0 +1,203 @@
+import sys
+import time
+import json
+from pathlib import Path
+from typing import Any, Dict
+
+sys.path.append('./')
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+from pose_format import Pose
+
+
+class SignLanguagePoseDataset(Dataset):
+    def __init__(self, data_dir: Path, split: str, disfluent_frame: int, fluent_frame: int, offset_frame: int = 1, dtype=np.float32, limited_num: int = -1):
+        """
+        Args:
+            data_dir (Path): Root directory where the data is saved. Each split should be in its own subdirectory.
+            split (str): Dataset split name, including "train", "validation", and "test".
+            disfluent_frame (int): Number of frames from the potentially disfluent sequence (updated pose sequence) to use as condition.
+            fluent_frame (int): Number of frames from the fluent (target) sequence (original pose sequence) to use as target.
+            offset_frame (int): Stride to use when generating sliding windows (default is 1).
+            dtype: Data type for the arrays, default is np.float32.
+            limited_num (int): Limit the number of samples to load; default -1 loads all samples.
+        """
+        self.data_dir = data_dir
+        self.split = split
+        self.disfluent_frame = disfluent_frame
+        self.fluent_frame = fluent_frame
+        self.window_size = disfluent_frame + fluent_frame
+        self.offset_frame = offset_frame
+        self.dtype = dtype
+
+        # Load all samples for the given split
+        # Each sample should have fluent (original), disfluent (updated), and metadata files
+        self.examples = []
+        split_dir = self.data_dir / split
+        fluent_files = sorted(list(split_dir.glob(f"{split}_*_original.pose")))
+        if limited_num > 0:
+            fluent_files = fluent_files[:limited_num]   # Limit the number of samples to load
+
+        frame_nums_list = []  # To record frame counts per sample
+        for fluent_file in fluent_files:
+            # Construct corresponding disfluent and metadata file paths based on the file name
+            disfluent_file = fluent_file.with_name(fluent_file.name.replace("_original.pose", "_updated.pose"))
+            metadata_file = fluent_file.with_name(fluent_file.name.replace("_original.pose", "_metadata.json"))
+            # Load pose objects
+            with open(fluent_file, 'rb') as f:
```

Comment:
here you load all of the files into memory. this means, that before any training can start, it first has to do a lot of work, and if the disk is slow or not enough memory, it will take a while or fail.
I recommend you have the examples as `fluent: Path, disfluent: Path, metadata: Path`
you **could** populate them as they come, or you could always load from disk

Reply:
Then, in `__getitem__` you can load from disk directly for each sample